init-distro-rid.sh

```diff
@@ -0,0 +1,178 @@
+            source $rootfsDir/android_platform
+            nonPortableBuildID="$RID"
+        fi
+
+        # RHEL 6 is the only distro we will check redHat release for.
+        if [ -e "${rootfsDir}/etc/os-release" ] && [ "${nonPortableBuildID}" == "" ] ; then
+            source "${rootfsDir}/etc/os-release"
+
+            # We have forced __PortableBuild=0. This is because -portablebuld
+            # has been passed as false.
+            if (( ${isPortable} == 0 )); then
+                if [ "${ID}" == "rhel" ]; then
+                    # remove the last version digit	
+                    VERSION_ID=${VERSION_ID%.*}
+                fi
+
+                nonPortableBuildID="${ID}.${VERSION_ID}-${buildArch}"
+            fi
+            
+        fi
+    fi
+
+    if [ "$buildOs" = "FreeBSD" ]; then
+        __freebsd_version=`sysctl -n kern.osrelease | cut -f1 -d'.'`
+        nonPortableBuildID="freebsd.$__freebsd_version-${buildArch}"
+    fi
+
+    if [ "${nonPortableBuildID}" != "" ]; then
+        export __DistroRid=${nonPortableBuildID}
+
+        # We are using a non-portable build rid. Force __PortableBuild to false.
+        export __PortableBuild=0
+    fi
+}
+
+
+# initDistroRidGlobal
+#
+# Input:
+#   os: (str)
+#   arch: (str)
+#   isPortable: (int)
+#   rootfsDir?: (nullable:string)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# The following out parameters are returned
+#
+#   __DistroRid
+#   __PortableBuild
+#
+initDistroRidGlobal()
+{
+    # __DistroRid must be set at the end of the function.
+    # Previously we would create a variable __HostDistroRid and/or __DistroRid.
+    #
+    # __HostDistroRid was used in the case of a non-portable build, it has been
+    # deprecated. Now only __DistroRid is supported. It will be used for both
+    # portable and non-portable rids and will be used in build-packages.sh
+
+    local buildOs=$1
+    local buildArch=$2
+    local isPortable=$3
+    local rootfsDir=$4
+
+    # Setup whether this is a crossbuild. We can find this out if rootfsDir
+    # is set. 
+    local isCrossBuild=0
+
+    if [ -z "${rootfsDir}" ]; then
+        isCrossBuild=0
+    else
+        # We may have a cross build. Check for the existance of the rootfsDir
+        if [ -e ${rootfsDir} ]; then
+            isCrossBuild=1
+        else
+            echo "Error rootfsDir has been passed, but the location is not valid."
+            exit 1
+        fi
+    fi
+
+    initNonPortableDistroRid ${buildOs} ${buildArch} ${isPortable} ${rootfsDir}
+
+    if [ -z "${__DistroRid}" ]; then
+        # The non-portable build rid was not set. Set the portable rid.
+
+        export __PortableBuild=1
+        local distroRid=""
+
+        # Check for alpine. It is the only portable build that will will have
+        # its name in the portable build rid.
+        if [ -e "${rootfsDir}/etc/os-release" ]; then
+            source "${rootfsDir}/etc/os-release"
+            if [ "${ID}" = "alpine" ]; then
```

Comment:
@am11 is there any way to distinguish between MUSL based Void Linux and GLIBC based one? We now detect only alpine at the portable build time to see if the RID should be linux-ARCH or linux-musl-ARCH. If there is a simple way to detect MUSL based system in generic way, it would be great.
@jashook I don't think we should block the merge on this, we can update that later. But I would probably forget about this if I didn't ask here.

Reply:
Yup, one approach we tried [in cmake](https://github.com/dotnet/corefx/issues/34316#issuecomment-451378638):
```sh
if ldd --version 2>&1 | grep -q musl; then echo "musl-libc is in use"; fi
```

Reply:
@am11 thank you, that sounds like a reasonable way to handle that.

Reply:
This should be added in a separate PR.